pkg/hubble/parser/seven/parser.go

```diff
@@ -134,7 +137,7 @@ func (p *Parser) Decode(r *accesslog.LogRecord, decoded *flowpb.Flow) error {
 	decoded.Type = flowpb.FlowType_L7
 	decoded.SourceNames = sourceNames
 	decoded.DestinationNames = destinationNames
-	decoded.L7 = decodeLayer7(r)
+	decoded.L7 = decodeLayer7(r, *p.opts)
```

Comment:
this reads a tad weird IMHO, maybe the pointer dance would be clearer if both `p.opts` and `args` (line 48) were not pointers.

Reply:
I think it's not a good idea to to not have them as pointers, cause we also pass them by reference when evaluating the `opts` at https://github.com/cilium/cilium/blob/e383fb572e0a3900887fac54c09f78b164083527/pkg/hubble/parser/seven/parser.go#L54.
It would be better if we pass the `p.opts` by reference here too like `decoded.L7 = decodeLayer7(r, p.opts)`. Looks more consistent and less ugly :). But let me know what you think.

Reply:
@ChrsMark yes I fully agree, to clarify I wasn't suggesting to change the functions themselves, but declare the variables (`p.opts` and `args`) as value as pass them as references (`&p.opts`, `&args`) when needed instead.

Reply:
current patch is great, resolving this conversation.